yarn-project/sequencer-client/src/sequencer/sequencer.ts

```diff
@@ -278,84 +282,43 @@ export class Sequencer extends (EventEmitter as new () => TypedEventEmitter<Sequ
       isPendingChainValid: pick(syncedTo.pendingChainValidationStatus, 'valid', 'reason', 'invalidIndex'),
     };
 
-    if (syncedTo.l1Timestamp + BigInt(this.l1Constants.ethereumSlotDuration) < ts) {
-      this.log.debug(
-        `Cannot propose block ${newBlockNumber} at next L2 slot ${slot} due to pending sync from L1`,
-        syncLogData,
-      );
-      return;
-    }
-
-    // Check that the slot is not taken by a block already
+    // Check that the slot is not taken by a block already (should never happen, since only us can propose for this slot)
     if (syncedTo.block && syncedTo.block.header.getSlot() >= slot) {
-      this.log.debug(
+      this.log.warn(
         `Cannot propose block at next L2 slot ${slot} since that slot was taken by block ${syncedTo.blockNumber}`,
         { ...syncLogData, block: syncedTo.block.header.toInspect() },
       );
       return;
     }
 
-    // Or that we haven't published it ourselves
-    if (this.lastBlockPublished && this.lastBlockPublished.header.getSlot() >= slot) {
-      this.log.debug(
-        `Cannot propose block at next L2 slot ${slot} since that slot was taken by our own block ${this.lastBlockPublished.number}`,
-        { ...syncLogData, block: this.lastBlockPublished.header.toInspect() },
-      );
-      return;
-    }
-
     // Check that we are a proposer for the next slot
-    let proposerInNextSlot: EthAddress | undefined;
-    try {
-      proposerInNextSlot = await this.epochCache.getProposerAttesterAddressInNextSlot();
```

Comment:
Heads up this line is changed within `checkCanPropose`: we now use `getProposerAttesterAddressIn(slot)` to avoid race conditions